converted module to pdk #107

enemarke opened the pull request:
Hi

I have converted the module to be pdk compliant and made a few minor refactors.

Some notes on what I have brooken and what is missing:

- The kitchen setup was removed in the pdk convertion scripts and Im not familiar with kitchen and have not added kitchen config.
- Readme has not been updated, since I would like some feedback on my work before updating it.

Cheers

mcgege commented:
Hi @enemarke , thanks a lot for working on this topic ... as I'm not the kitchen guru, maybe @artem-sidorenko can help us here?

I also would suggest to separate the functional optimization (`/lib/*`, `/manifests/*`) from the pdk / testing changes. Can you put the first block in a separate pr? I could test & merge it prior to this ...

mcgege commented on Rakefile:
```diff
@@ -1,71 +1,4 @@
-
-  task :default => [:run_all_linters, :spec]
-
-  # Changelog Generator
```
I definitely need this one for generating the Changelog

mcgege commented on metadata.json:
```diff
@@ -72,5 +66,13 @@
+      "version_requirement": ">= 4.0.0 < 6.0.0"
+    }
+  ],
+  "template-url": "file:///opt/puppetlabs/pdk/share/cache/pdk-templates.git",
```
Maybe we should change this to `https://github.com/puppetlabs/pdk-templates`

enemarke replied:
Done

enemarke commented:
I have made a new commit that removed changes to /lib/*, /manifest/* and some in /spec/. I hope this is the correct way for moving changes into a new pull request.

artem-sidorenko commented on .travis.yml:
```diff
@@ -1,40 +1,45 @@
+---
 sudo: false
+dist: trusty
```
this is not needed anymore, its a new default

enemarke commented:
@artem-sidorenko That makes sense. Should I close this pull request and open one or two new ones in order to get fewer cleaner commits or should I just make changes directly to this pull request?

artem-sidorenko commented:
@enemarke its up to you, you can also cleanup via [rebase](https://git-scm.com/book/en/v2/Git-Branching-Rebasing), squash the commits and force push your changes

enemarke commented:
I have rebased into fewer commits and removed changes to travis and test-kitchen. I tried testing with kitchen on my laptop by it was unsuccessful. Can someone verify that test-kitchen configs are correct?s

artem-sidorenko commented:
I pushed it for integration tests: https://travis-ci.org/dev-sec/puppet-os-hardening/builds/341803747, it looks good to me. The failing tests are not fixed yet.

@enemarke could you maybe squash all other commits to one "pdk" commit?
@mcgege LGTM, your turn ;-)

mcgege commented:
@enemarke Also looks good to me ... could you please rebase your changes now on top of the master branch so that we could merge it?

enemarke commented:
Rebasing done.

mcgege commented:
@enemarke Retested it, looks good ... one final wish: Could you please [rewrite your commit message](https://help.github.com/articles/changing-a-commit-message/), compact it and remove those comments, that have already been merged before?

enemarke commented:
@mcgege I have rewritten the commit message.

mcgege commented:
@enemarke Thanks a lot!
